addons/web/static/tests/views/fields/timezone_mismatch_field_tests.js

```diff
@@ -0,0 +1,175 @@
+            'Should display the datetime in the selected timezone'
+        );
+    });
+
+    QUnit.test('widget timezone_mismatch in a form view', async function (assert) {
+        assert.expect(2);
+
+        const serverData = {
+            models: {
+                partner: {
+                    fields: {
+                        tz_offset: {
+                            string: 'tz_offset',
+                            type: 'char',
+                        },
+                        tz: {
+                            type: 'selection',
+                            selection: [
+                                ['Europe/Brussels', 'Europe/Brussels'],
+                                ['America/Los_Angeles', 'America/Los_Angeles'],
+                            ],
+                        },
+                    },
+                    records: [{ id: 1, tz: false, tz_offset: '+4800' }],
+                },
+            },
+        };
+
+        await makeView({
+            type: 'form',
+            resModel: 'partner',
+            serverData,
+            resId: 1,
+            arch: /*xml*/ `
+                <form>
+                    <field name="tz_offset" invisible="True"/>
+                    <field name="tz" widget="timezone_mismatch" />
+                </form>
+            `,
+        });
+
+        assert.containsOnce(target, 'div[name="tz"]>select');
+
+        const $timezoneMismatch = $(target).find('.o_tz_warning');
+        assert.strictEqual($timezoneMismatch.length, 1, 'warning class should be there.');
+    });
+
+    QUnit.test('widget timezone_mismatch in a form view edit mode with mismatch', async function (assert) {
+        assert.expect(3);
+
+        const serverData = {
+            models: {
+                partner: {
+                    fields: {
+                        tz_offset: {
+                            string: 'tz_offset',
+                            type: 'char',
+                        },
+                        tz: {
+                            type: 'selection',
+                            selection: [
+                                ['Europe/Brussels', 'Europe/Brussels'],
+                                ['America/Los_Angeles', 'America/Los_Angeles'],
+                            ],
+                        },
+                    },
+                    records: [{ id: 1, tz: 'America/Los_Angeles', tz_offset: '+4800' }],
+                },
+            },
+        };
+
+        await makeView({
+            type: 'form',
+            resModel: 'partner',
+            serverData,
+            resId: 1,
+            arch: /*xml*/ `
+                <form>
+                    <field name="tz_offset" invisible="True"/>
+                    <field name="tz" widget="timezone_mismatch" options="{\'tz_offset_field\': \'tz_offset\'}"/>
+                </form>
+            `,
+        });
+
+        const $timezoneEl = $(target).find('div[name="tz"]>select');
```

Comment:
I have to change this because the "select" element is inside a "div" with name="tz". The original selector is `'select[name="tz"]'`.